packages/clay-core/src/tree-view/TreeViewItem.tsx

```diff
@@ -565,7 +565,8 @@ export function TreeViewItemStack({
 	const nestedChildren =
 		nestedKey && (item[nestedKey] as Array<Record<string, any>>);
 
-	const hasChildren = nestedChildren && nestedChildren.length > 0;
+	const hasChildren =
+		(nestedChildren && nestedChildren.length > 0) || item.hasChildren;
```

Comment:
Hey @sandrodw3 it's a good idea, I think we can do it in another way so as not to force a property called `hasChildren` to exist in the tree data, I think we can add an API for `TreeView.Item` and `TreeView.ItemStack` this API with that name, so you can use any other data to identify that there are children in the item.

```jsx
<TreeView>
  {(item) => (
    <TreeView.Item expandable={item.children.length}>
      {item.name}
    </TreeView.Item>
  )}
</TreeView>
```

We can use an API that we use internally to make this `expandable` and make it public, I think we also need to improve how this is being done now because when there is no children, we are still rendering a button with no content when the `loadMore` is added, I think this was done wrong, I will send some commits on top of your PR to improve this too and make this logic simpler.

Reply:
Hey @matuzalemsteles that's definitely a better solution, thanks! 🙂